File: 백준/Bronze/2231. 분해합/분해합.cc
#include <iostream>

using namespace std;

int main()
{
    string input;
    int ans, check;
    
    cin >> input;
    int n = atoi(input.c_str());
    
    for(int i = 1; i <= n; i++)
    {
        check = i;
        string k = to_string(i);
        
        for(int j = 0; j < k.length(); j++)
        {
            
           check += k[j] - '0';
        }
        
        if(n == check)
        {
            ans = i;
            cout << ans;
            break;
        }
        
        
    }
    
    if(n != check) cout << "0";
    
    return 0;
}

File: 백준/Bronze/2231. 분해합/분해합.kt
fun main() {

    val input = readln().toInt()
    for (no in 1..input) {
        var tempAns = no
        no.toString().forEach {
            tempAns += it - '0'
        }
        if (input == tempAns) {
            println(no)
            return
        }
    }
    println(0)

}
